modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache

```diff
@@ -385,20 +385,32 @@ export class {{classname}} {
 
 {{/hasFormParams}}
 {{#useHttpClient}}
+    {{^isResponseFile}}
+        let responseType: string = undefined;
+        if(httpHeaderAcceptSelected !== undefined) {
+            responseType = httpHeaderAcceptSelected.startsWith('text') ? 'text' : 'json';
+        }
+
+    {{/isResponseFile}}
+        const httpClientParams: any = {
+    {{#hasQueryParams}}
+            params: queryParameters,
+    {{/hasQueryParams}}
+    {{#isResponseFile}}
+            responseType: "blob",
+    {{/isResponseFile}}
+    {{^isResponseFile}}
+            responseType: responseType,
+    {{/isResponseFile}}
+            withCredentials: this.configuration.withCredentials,
+            headers: headers,
+            observe: observe,
+            reportProgress: reportProgress
+        }
+
         return this.httpClient.{{httpMethod}}{{^isResponseFile}}<{{#returnType}}{{{returnType}}}{{#isResponseTypeFile}}|undefined{{/isResponseTypeFile}}{{/returnType}}{{^returnType}}any{{/returnType}}>{{/isResponseFile}}(`${this.configuration.basePath}{{{path}}}`,{{#isBodyAllowed}}
             {{#bodyParam}}{{paramName}}{{/bodyParam}}{{^bodyParam}}{{#hasFormParams}}convertFormParamsToString ? formParams.toString() : formParams{{/hasFormParams}}{{^hasFormParams}}null{{/hasFormParams}}{{/bodyParam}},{{/isBodyAllowed}}
-            {
-{{#hasQueryParams}}
```

Comment:
I suggest to leave the inline params here instead of using an object with type any